core/src/main/java/com/graphhopper/routing/util/parsers/OSMRoadAccessParser.java

```diff
@@ -57,11 +57,12 @@ public IntsRef handleWayTags(IntsRef edgeFlags, ReaderWay readerWay, boolean fer
             }
         }
 
-        SpatialRuleSet spatialRuleSet = readerWay.getTag("spatial_rule_set", null);
-        if (spatialRuleSet != null && spatialRuleSet != SpatialRuleSet.EMPTY) {
-            RoadClass roadClass = RoadClass.find(readerWay.getTag("highway", ""));
-            accessValue = spatialRuleSet.getAccess(roadClass, TransportationMode.CAR, YES);
-        }
+        RoadClass roadClass = RoadClass.find(readerWay.getTag("highway", ""));
+        // todo: for spatial rules we got the entire set of rules for this way, but now we throw an error in case
+        //       there are multiple rules for the same way. is this ok?
+        CountryRule countryRule = readerWay.getTag("country_rule", null);
```

Comment:
The SpatialRule interface is more generic. You could have multiple rules with different areas which target the same way. e.g country/state/area. Our previous country rules are just a special case which happens to be overlapping free.

But for dedicated CountryRules we can give that guarantee.

The usecase for a set of rules would be point 5 of your comment: https://github.com/graphhopper/graphhopper/issues/2246#issuecomment-870096236. With custom areas there is no such guarantee that the borders don't overlap. We should keep this in mind if we tackle those later on.

Reply:
> But for dedicated CountryRules we can give that guarantee.

Ok yes, because our countries.geojson has this property.

> With custom areas there is no such guarantee that the borders don't overlap. We should keep this in mind if we tackle those later on.

The custom areas are already added as artificial/temporary way tag by the OSMReader, so the complete `List<CustomArea>` is available to all tag parsers and flag encoders. I don't think there is anything else we need to do there and the rest is up to the single tag parsers (they can either throw an error if multiple matches were not expected or make use of all matches etc.).

Reply:
I'll remove the this todo, because like you said there is only one country match.